app/Http/Controllers/V2/Polygons/ChangeStatusPolygonsController.php

```diff
@@ -14,19 +14,22 @@ class ChangeStatusPolygonsController extends Controller
     public function __invoke(ChangeStatusPolygonsUpdateRequest $request)
     {
         $body = $request->all();
-        $polygons = $body['updatePolygons'];
+        $updatePolygons = $body['updatePolygons'];
 
-        foreach ($polygons as $polygon) {
-            $sitePolygon = SitePolygon::where('uuid', $polygon['uuid'])->first();
-            if (! $sitePolygon) {
+        $sitePolygons = SitePolygon::whereIn('uuid', collect($updatePolygons)->map(fn ($p) => $p['uuid']))->get();
+        $changedPolygons = [];
+        foreach ($sitePolygons as $sitePolygon) {
+            $foundPolygon = collect($updatePolygons)->first(fn ($p) => $p['uuid'] === $sitePolygon->uuid);
```

Comment:
You're collecting the polygons many times. Would be better to do it once on line 17 and use the collection everywhere else.